### PR TITLE
MultiChannels for tvOS player

### PR DESCRIPTION
**Pull request by sheiladoherty-dolby**
This is a basic implementation of Multi-Channels for tvOS. There are various features such as labels, sound management & persistence, that need to by firmed up in a later ticket.

- TabView for using single play vs multichannel
- ChannelView can take up to 4 inputs
- for ease of testing, the 4 streams are currently pre-populated
- if the textfields are empty, they will just be skipped, up to 4 players

LandingView
Simulator Screenshot - Apple TV 4K (3rd generation) - 2024-09-23 at 13 57 42

2 Channels
Simulator Screenshot - Apple TV 4K (3rd generation) - 2024-09-23 at 13 57 21

3 Channels
Simulator Screenshot - Apple TV 4K (3rd generation) - 2024-09-23 at 13 57 51

4 Channels
Simulator Screenshot - Apple TV 4K (3rd generation) - 2024-09-19 at 13 58 26

### CODE REVIEW

File: rts-viewer-tvos/RTSViewer/ChannelGridView/ChannelGridView.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                            }
+                        }
+                        .onDisappear {
+                            ChannelGridViewModel.logger.debug("♼ Channel Grid view: Video view disappear for \(source.sourceId)")
+                            Task {
+                                await channel.videoTracksManager.disableTrack(for: source, on: viewId)
+                            }
+                        }
+                        .id(source.id)
+                        .id(channel.source.id)
```
Comment (aravind-raveendran) [inline rts-viewer-tvos/RTSViewer/ChannelGridView/ChannelGridView.swift:56]: the second `id` view modifier is redundant?

File: rts-viewer-tvos/RTSViewer/ChannelGridView/ChannelGridViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+//  ChannelGridViewModel.swift
+//
+
+import Foundation
+import MillicastSDK
+import os
+import SwiftUI
+
+@MainActor
+final class ChannelGridViewModel: ObservableObject {
```
Comment (aravind-raveendran) [inline rts-viewer-tvos/RTSViewer/ChannelGridView/ChannelGridViewModel.swift:11]: if no mutations it does not need to be ObservableObject and can be a struct?

File: rts-viewer-tvos/RTSViewer/ChannelView/ChannelViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+}
+
+private extension ChannelViewModel {
+    // swiftlint:disable function_body_length cyclomatic_complexity
+    func startObservers() {
+        Task { [weak self] in
+            guard let self,
+                  let channels else { return }
+
+            for channel in channels {
```
Comment (aravind-raveendran) [inline rts-viewer-tvos/RTSViewer/ChannelView/ChannelViewModel.swift:91]: Minor: Do you think some of subscriptions(iterations) can be setup concurrently ?

File: rts-viewer-tvos/RTSViewer/ChannelView/ChannelViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            }
+        }
+    }
+
+    // swiftlint:enable function_body_length cyclomatic_complexity
+
+    func updateChannelWithSources(channel: Channel, sources: [StreamSource]) {
+        guard !sourcedChannels.contains(where: { $0.id == channel.id }),
+              sources.count > 0 else { return }
+        let sourcedChannel = SourcedChannel.build(from: channel, source: sources[0])
```
Comment (aravind-raveendran) [inline rts-viewer-tvos/RTSViewer/ChannelView/ChannelViewModel.swift:162]: not sure if we should use zeroth index or source which is `main`? the latter is what the tvOS app currently does and an easy use case to handle. 

We also need to handle removal when the source goes inactive in this case will be removed from the list of sources?

File: rts-viewer-tvos/RTSViewer/Managers/VideoTracksManager.swift [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+//  VideoTracksManager.swift
+//
+
+import Combine
+import Foundation
+import MillicastSDK
+import os
+import RTSCore
+
+final actor VideoTracksManager {
```
Comment (aravind-raveendran) [inline rts-viewer-tvos/RTSViewer/Managers/VideoTracksManager.swift:11]: I am not quite sure, my gut feel is this track manager may not be required for this use case, in interactive player we have scenarios where multiple views will try to play the same video (in a detail view vs grid/list layout). Currently our SDK has a limitation where only one renderer can be active and attached to a track at a given time.
So every time the user navigate back from the detail view the previous renderer is used reenable the track. 

In this multichannel app - all we need will be a layer observation and enable track(with a newly selected layer) ? What's your thoughts?

File: rts-viewer-tvos/RTSViewer/Models/VideoQuality.swift [hunk 6]
```diff
@@ -24,6 +15,26 @@ enum VideoQuality: Identifiable {
 }
 
 extension VideoQuality {
+    static func < (lhs: VideoQuality, rhs: VideoQuality) -> Bool {
+        return switch (lhs, rhs) {
+        case(.quality, .auto):
+            false
+        case let (.quality(lhsQuality), .quality(rhsQuality)):
+            lhsQuality == rhsQuality
```
Comment (aravind-raveendran) [inline rts-viewer-tvos/RTSViewer/Models/VideoQuality.swift:23]: do we use this operator ?  `lhsQuality == rhsQuality` can be a false comparison for less than ?

**Comment by sheiladoherty-dolby** [comment]
Adding more features & will resubmit

**Pull request closed without merging**